contracts/contracts/rem.swap/include/rem.swap/rem.swap.hpp

```diff
@@ -0,0 +1,210 @@
+/**
+ *  @copyright defined in eos/LICENSE.txt
+ */
+
+#pragma once
+
+#include <eosio/action.hpp>
+#include <eosio/asset.hpp>
+#include <eosio/crypto.hpp>
+#include <eosio/eosio.hpp>
+#include <eosio/permission.hpp>
+#include <eosio/privileged.hpp>
+#include <eosio/time.hpp>
+#include <eosio/system.hpp>
+
+#include <vector>
+#include <string>
+
+namespace eosiosystem {
+    class system_contract;
+}
+
+namespace eosio {
+
+    using std::string;
+
+    /**
+     * @defgroup remswap rem.swap
+     * @ingroup eosiocontracts
+     *
+     * rem.swap contract
+     *
+     * @details rem.swap contract defines the structures and actions that allow users to init token swap,
+     * finish token swap, create new account, init token swap to ERC20 - for REM tokens.
+     * @{
+     */
+    class [[eosio::contract("rem.swap")]] swap : public contract {
+    public:
+        using contract::contract;
+
+        /**
+         * Init swap action.
+         *
+         * @details Initiate token swap with ERC20 REM ( ERC20 - > Remchain ).
+         *
+         * @param user - the owner account to execute the init action for.
+         * @param swap_key - the public key created for the token swap.
+         * @param trx_id - the transaction transfer id in Ethereum blockchain.
+         * @param quantity - the quantity of tokens in the transfer transaction.
+         * @param swap_init_time - the timestamp transfer transaction in Ethereum blockchain.
+         */
+        [[eosio::action]]
+        void init( const name& user, public_key swap_key, const string& trx_id,
+                   asset& quantity, const time_point& swap_init_time );
+
+        /**
+         *  Debug action.
+         */
+        [[eosio::action]]
+        void cleartable( const name& user );
+
+
+        /**
+         * Finish token swap with ERC20 REM.
```

Comment:
ERC20 is not the only source of swaps, let's not mention it explicitly without a need

Reply:
@roman-tik, done.